### PR TITLE
Added the periodic Sync

### PR DESCRIPTION
**Pull request by pierrelabat**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/os/hsestore/HseStore.cc [hunk 1]
```diff
@@ -167,41 +167,65 @@ void HseStore::Collection::committed_wait_persist_cb(HseStore::Collection *c)
...
+  HseStore::Collection *c;
+  bool needsync = false;
+
+  // Record for each collection, the latest committed before starting the sync.
+  boost::shared_lock<ceph::shared_mutex> l{store->coll_lock};
+  for (auto const& [key, val] : store->coll_map) {
+		c = static_cast<HseStore::Collection*>(val.get());
+		c->_t_seq_committed_wait_sync = c->_t_seq_next - 1;
+		if (c->_t_seq_persisted_latest < c->_t_seq_committed_wait_sync)
+		  needsync = true;
```
Comment (tristan957) [inline src/os/hsestore/HseStore.cc:182]: indentation

File: src/os/hsestore/HseStore.cc [hunk 2]
```diff
@@ -167,41 +167,65 @@ void HseStore::Collection::committed_wait_persist_cb(HseStore::Collection *c)
...
 //
 
+// Sync the whole kvdb
+void Syncer::kvdb_sync(HseStore *store)
+{
+  HseStore::Collection *c;
+  bool needsync = false;
+
+  // Record for each collection, the latest committed before starting the sync.
+  boost::shared_lock<ceph::shared_mutex> l{store->coll_lock};
```
Comment (tristan957) [inline src/os/hsestore/HseStore.cc:177]: Use std::shared_lock. We can go back and fix the other places in the file later to be conssitent.

File: src/os/hsestore/HseStore.cc [hunk 3]
```diff
@@ -167,41 +167,65 @@ void HseStore::Collection::committed_wait_persist_cb(HseStore::Collection *c)
...
+		c->_t_seq_committed_wait_sync = c->_t_seq_next - 1;
+		if (c->_t_seq_persisted_latest < c->_t_seq_committed_wait_sync)
+		  needsync = true;
+  }
+  l.unlock();
+
+  if (!needsync)
+    return;
+
+  hse_kvdb_sync(store->kvdb);
```
Comment (tristan957) [inline src/os/hsestore/HseStore.cc:189]: If the sync fails, should we log anything or abort?

Comment (pierrelabat) [reply src/os/hsestore/HseStore.cc:189]: Added log.

File: src/os/hsestore/HseStore.cc (hunk 1 above)
Comment (pierrelabat) [reply src/os/hsestore/HseStore.cc:182]: fixed

File: src/os/hsestore/HseStore.cc (hunk 2 above)
Comment (pierrelabat) [reply src/os/hsestore/HseStore.cc]: Changed.

**Pull request merged**